model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/entities/CachedRealm.java

```diff
@@ -241,8 +241,7 @@ public CachedRealm(Long revision, RealmModel model) {
         requiredCredentials = model.getRequiredCredentialsStream().collect(Collectors.toList());
         userActionTokenLifespans = Collections.unmodifiableMap(new HashMap<>(model.getUserActionTokenLifespans()));
 
-        this.identityProviders = model.getIdentityProvidersStream().map(IdentityProviderModel::new)
-                .collect(Collectors.toList());
+        this.identityProviders = model.getIdentityProvidersStream().collect(Collectors.toList());
```

Comment:
Yeah, it is not correct because here we are caching the model by creating another instance and copying data over.

We can copy the icon class here https://github.com/keycloak/keycloak/blob/76d9125c3fad117a23d87cd6e6b054007c6770cf/server-spi/src/main/java/org/keycloak/models/IdentityProviderModel.java#L84 and keep it cached.

I'm not sure why there is no field set to `displayIconClasses` but it should be OK to have one.

Reply:
True.  
Thx for your input. Feels way simpler now :)